packages/flutter_tools/test/general.shard/runner/runner_test.dart

```diff
@@ -90,6 +90,7 @@ void main() {
     }, overrides: <Type, Generator>{
       Platform: () => FakePlatform(environment: <String, String>{
         'FLUTTER_ANALYTICS_LOG_FILE': 'test',
+        'FLUTTER_DOCTOR_HOST_TIMEOUT': '0',
```

Comment:
Can these tests inject an http client factory that gives a `FakeHttpClient` that causes the host validator to pass?